services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -42,8 +55,12 @@ func NewProducer(destinationConfig interface{}) (*PubsubClient, error) {
 		return nil, fmt.Errorf("[GooglePubSub] error  :: error in GooglePubSub while unmarshelling destination config:: %w", err)
 	}
 	var client *pubsub.Client
-	if config.Credentials != "" && config.ProjectId != "" {
-		client, err = pubsub.NewClient(ctx, config.ProjectId, option.WithCredentialsJSON([]byte(config.Credentials)))
+	if config.ProjectId != "" && (config.Credentials != "" || config.TestConfig.Endpoint != "") {
+		if config.Credentials != "" {
```

Comment:
[minor] this section of the code is kinda hard to read, any refactoring and/or documentation would be helpful.